### ISSUE TITLE
proposed API

### ISSUE THREAD

**Issue by brentp** [brentp/libbigwig-nim#1]
for comments / feedback.
see:
+ [pybigwig](https://github.com/deeptools/pyBigWig#retrieve-values-for-individual-bases-in-a-range)
+ [bw-python](https://github.com/brentp/bw-python)

```Nim
var bw: BigWig
bw.open(path, fmRead)

# avoid allocating when possible
var values: seq[float32]
bw.values(values, "chr1", 0, 2222)

for iv in bw.intervals("chr2", 999, 88888): # iterator.
    ## tuple[start: int, stop: int, value: float32]

# single value
var m = bw.stats("chr2", 999, 9999, stat=Stat.mean) 

# broken into 10 bins. Stat is an enum with mean, coverage, sum, std, etc.
var ameans = newSeq[float32](10)
# if start and stop are not given, defaults to entire chrom
bw.stats("chr2", start=999, stop=9999, ameans, stat=Stat.mean) 
```

once this simple reading interface is complete, I'll open another issue for writing and header manipulation, likely following [this](https://github.com/deeptools/pyBigWig#add-a-header-to-a-bigwig-file) once reading is complete

**Comment by brentp** [comment]
this is implemented mostly as described here and exactly as described in the README
